Safety/AttitudeManager/Src/attitudeStateClasses.cpp

```diff
@@ -223,3 +246,79 @@ attitudeState& FatalFailureMode::getInstance()
     return singleton;
 }
 
+void DisarmMode:: execute(attitudeManager* attitudeMgr)
+{
+    //setting PWM channel values to lowest "disarm" value
+    attitudeMgr->pwm->set(FRONT_LEFT_MOTOR_CHANNEL, 0);
+    attitudeMgr->pwm->set(FRONT_RIGHT_MOTOR_CHANNEL, 0);
+    attitudeMgr->pwm->set(BACK_LEFT_MOTOR_CHANNEL, 0);
+    attitudeMgr->pwm->set(BACK_RIGHT_MOTOR_CHANNEL, 0);
+
+    const uint8_t TIMEOUT_THRESHOLD = 2; //Max cycles without data until connection is considered broken
+
+    //Get Arm Disarm instruction
+    if(ReceiveArmDisarmInstruction(attitudeMgr))
+    {
+        armDisarmTimeoutCount = 0;
+    }
+    else
+    {
+        if(armDisarmTimeoutCount < TIMEOUT_THRESHOLD)
+            armDisarmTimeoutCount++;
+    }
+
+    /*
+        3 possibilities:
+             1. Go into FatalFailureMode bec of timeout
+             2. Go into fetchInstructionsMode bec "Arm" instruction was sent
+             3. Do nothing, stay in the disarm state
+    */
+    if(armDisarmTimeoutCount > TIMEOUT_THRESHOLD && CommsFailed())
+    {
+        //Abort due to timeout failures
+        attitudeMgr->setState(FatalFailureMode::getInstance());
+        return;
```

Comment:
It seems to me that this code will basically boot the pilot into the fatalFailureMode if they are disarmed & disconnected. This seems rather odd to me? What if the pilot turns on the transmitter after they turn on the drone? What if it takes longer for the receiver to boot up then the safety chip takes to run these two iterations (not that I'm expecting that)? In my mind the disarm state would be the ONLY time that you would want to be able to disconnect the transmitter, all other times should lockout. Thoughts @upadhyaydhruv @antholuo?

Reply:
If comms failed, would we be able to know if we have changed from arm to disarm?

Reply:
After talking with Anthony, I'm okay to leave this as is. Marking as resolved.